### PR TITLE
Initial test runner to hackathon dev

### PR DESCRIPTION
**Pull request by richardwerkman**

### CODE REVIEW

File: Stryker.NET/Stryker.NET/IsolatedRunners/TestRunner.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            _tempDirectory = $"{_rootDirectory}\\stryker_temp";
+            _command = "dotnet";
+            
+        }
+
+        public void Test()
+        {
+            var arguments = $"test";
+            RunCommand(arguments);
+
```
Comment (yaelkeemink) [inline Stryker.NET/Stryker.NET/IsolatedRunners/TestRunner.cs]: Empty line

File: Stryker.NET/Stryker.NET/IsolatedRunners/TestRunner.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            process.ErrorDataReceived += Process_ErrorDataReceived;
+            process.BeginOutputReadLine();
+            process.BeginErrorReadLine();
+            process.WaitForExit();
+        }
+
+        private void Process_ErrorDataReceived(object sender, DataReceivedEventArgs e)
+        {
+            //ignore
+            //System.Console.WriteLine(e.Data);
```
Comment (yaelkeemink) [inline Stryker.NET/Stryker.NET/IsolatedRunners/TestRunner.cs]: Remove these comments

File: Stryker.NET/Stryker.NET/IsolatedRunners/TestRunner.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        }
+
+        private void Process_OutputDataReceived(object sender, DataReceivedEventArgs e)
+        {
+            //ignore
+            //System.Console.WriteLine(e.Data);
```
Comment (yaelkeemink) [inline Stryker.NET/Stryker.NET/IsolatedRunners/TestRunner.cs]: Remove these comments

File: Stryker.NET/Stryker.NET/Managers/DirectoryManager.cs [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            foreach (string newPath in Directory.GetFiles(source, "*.*",
+                SearchOption.AllDirectories).Where(a => !a.StartsWith("Db") && !a.EndsWith(".lock")))
+            {
+                File.Copy(newPath, newPath.Replace(source, destination), true);
+            }
+        }
+
+        public IEnumerable<string> GetFiles(string source)
+        {
+            var options = new StykerOptions();
```
Comment (yaelkeemink) [inline Stryker.NET/Stryker.NET/Managers/DirectoryManager.cs]: typo ''stRykerOptions"

File: Stryker.NET/Stryker.NET/Managers/DirectoryManager.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+            var reflector = new FileReflector(options, source);
+            return reflector.GetFilesToMutate();
+        }
+
+        public void Dispose()
+        {
+            if (Directory.Exists(_tempDirectory))
```
Comment (yaelkeemink) [inline Stryker.NET/Stryker.NET/Managers/DirectoryManager.cs]: Is this the place wher we want it to be? Or do we want this to be a method of the DirectoryManager and being called from the dispose of stryker?

File: Stryker.NET/Stryker.NET/Reporters/CleartTextReporter.cs [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+          Mutator: BinaryOperator
+          -                 return user.age >= 18;
+          +                 return user.age > 18;
+
+          Mutant survived!
+          /yourPath/yourFile.js: line 10:27
+          Mutator: RemoveConditionals
+          -                 return user.age >= 18;
+          +                 return true;
+        */
```
Comment (yaelkeemink) [inline Stryker.NET/Stryker.NET/Reporters/CleartTextReporter.cs]: Remove this comment

File: Stryker.NET/Stryker.NET/Reporters/CleartTextReporter.cs [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        public void Report(Mutant mutant)
```
Comment (yaelkeemink) [inline Stryker.NET/Stryker.NET/Reporters/CleartTextReporter.cs]: Use $@

File: Stryker.NET/Stryker.NET/Stryker.cs [hunk 8]
```diff
@@ -1,27 +1,75 @@
...
-                
+                var path = mutant.FilePath;
+
+                // (over)write temp code file
+                File.WriteAllText(path, mutant.MutatedCode, Encoding.Unicode);
+
+                // run unit tests with mutant
+                _testRunner.Test();
+
+                _reporter?.Report(mutant);
```
Comment (yaelkeemink) [inline Stryker.NET/Stryker.NET/Stryker.cs]: Do we want to proceed when there is no reporter?

File: Stryker.NET/Stryker.NET/Reporters/CleartTextReporter.cs (hunk 7 above)
Comment (Razsiel) [reply Stryker.NET/Stryker.NET/Reporters/CleartTextReporter.cs]: Using a Stringbuilder is both more readable, efficient (less so than string interpolation afaik, but efficient enough) and has the benefit of not being tab sensitive like $@ is.

File: Stryker.NET/Stryker.NET/IsolatedRunners/TestRunner.cs [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    public class TestRunner : ITestRunner
+    {
+        private readonly string _rootDirectory;        
+        private readonly string _tempDirectory;
+        private readonly string _command;
+
+        public TestRunner(string rootDir)
+        {
+            _rootDirectory = rootDir;
+            _tempDirectory = $"{_rootDirectory}\\stryker_temp";
```
[the remark this answers is not in the document]
Comment (Razsiel) [reply Stryker.NET/Stryker.NET/IsolatedRunners/TestRunner.cs]: Will be fixed in next pull request

**Pull request merged**